cinema/views.py

```diff
@@ -0,0 +1,65 @@
+from django.shortcuts import get_object_or_404
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from cinema.models import Movie
+from cinema.serializers import MovieSerializer
+
+
+class MovieCRUDView(APIView):
+    model = Movie
+    serializer_class = MovieSerializer
+
+    def get_object(self, pk: int):
+        instance = get_object_or_404(self.model, pk=pk)
+        return instance
+
+    def get(self, request, *args, **kwargs):
+        pk = kwargs.get("pk")
+        if pk is not None:
+            movie = self.get_object(pk)
+            serializer = self.serializer_class(movie)
+        else:
+            movies = self.model.objects.all()
+            serializer = self.serializer_class(movies, many=True)
+        return Response(serializer.data, status.HTTP_200_OK)
+
+    def post(self, request, *args, **kwargs):
+        title = request.data.get("title")
+        if self.model.objects.filter(title=title).exists():
+            return Response(
+                {"error": f"Object with title '{title}' is already exists"},
+                status=status.HTTP_400_BAD_REQUEST
+            )

```

Comment:
Validation is redundant, just change into model unique=True

Reply:
Serializers does not handle UNIQUE errors by default